tools/kani-cov/src/coverage.rs

```diff
@@ -0,0 +1,209 @@
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum CheckStatus {
+    Failure,
+    Covered,   // for `code_coverage` properties only
+    Satisfied, // for `cover` properties only
+    Success,
+    Undetermined,
+    Unreachable,
+    Uncovered,     // for `code_coverage` properties only
+    Unsatisfiable, // for `cover` properties only
+}
+
+/// Kani coverage check.
+///
+/// Note: This data structure should not be duplicated in Kani -
+/// <https://github.com/model-checking/kani/issues/3541>
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CoverageCheck {
+    pub function: String,
+    term: CoverageTerm,
+    pub region: CoverageRegion,
+    pub status: CheckStatus,
+}
+
+// Note: This `impl` should not be duplicated in Kani -
+// <https://github.com/model-checking/kani/issues/3541>
+impl std::fmt::Display for CheckStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let check_str = match self {
+            CheckStatus::Satisfied => style("SATISFIED").green(),
+            CheckStatus::Success => style("SUCCESS").green(),
+            CheckStatus::Covered => style("COVERED").green(),
+            CheckStatus::Uncovered => style("UNCOVERED").red(),
+            CheckStatus::Failure => style("FAILURE").red(),
+            CheckStatus::Unreachable => style("UNREACHABLE").yellow(),
+            CheckStatus::Undetermined => style("UNDETERMINED").yellow(),
+            CheckStatus::Unsatisfiable => style("UNSATISFIABLE").yellow(),
+        };
+        write!(f, "{check_str}")
+    }
+}
+
+/// Raw Kani coverage results.
+///
+/// Note: This data structure should not be duplicated in Kani -
+/// <https://github.com/model-checking/kani/issues/3541>
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CoverageResults {
+    pub data: BTreeMap<String, Vec<CoverageCheck>>,
+}
+
+/// Aggregated coverage results.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CombinedCoverageResults {
+    pub data: BTreeMap<String, Vec<(String, Vec<CovResult>)>>,
+}
+
+/// The coverage result associated to a particular coverage region.
+/// Basically, this aggregates the information of one or more `CoverageCheck`.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CovResult {
+    pub function: String,
+    pub region: CoverageRegion,
+    pub times_covered: u32,
+    pub total_times: u32,
+}
+
+/// A coverage region.
+///
+/// Note: This data structure should not be duplicated in Kani -
+/// <https://github.com/model-checking/kani/issues/3541>
+#[derive(Debug, Clone, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct CoverageRegion {
+    pub file: String,
+    pub start: (u32, u32),
+    pub end: (u32, u32),
+}
+
+impl Display for CoverageRegion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}:{} - {}:{}", self.start.0, self.start.1, self.end.0, self.end.1)
```

Comment:
Maybe let's include the file name as well.
```suggestion
        write!(f, "{}:{}:{} - {}:{}", self.file, self.start.0, self.start.1, self.end.0, self.end.1)
```

Reply:
Ok, done.